// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.first_grpc_project";
option java_outer_classname = "FirstGrpcProjectProto";
option objc_class_prefix = "HLW";

package first_grpc_project;

// The greeting service definition.

service Adder {

  rpc login(loginRequest) returns (loginResponse) {}

  // Sends an addRequest
  rpc add (addRequest) returns (addResponse) {}

  rpc getMultiplicationTable(tableRequest) returns (stream tableResponse){}

}

message loginRequest {
  string username = 1;
  string password = 2;
}

message loginResponse {
  string jwt_token = 1;
}

// The request message containing the user's name.
message addRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}
   
// The response message containing the greetings
message addResponse {
  int32 result = 1;
}


//multiplication request
message tableRequest{
  int32 num = 1;
  int32 n = 2;
}

//multiplication response
message tableResponse{
  int32 num = 1;
  int32 n = 2;
  int32 result = 3;
}



